src/groupeddataframe/groupeddataframe.jl

```diff
@@ -572,6 +603,14 @@ function Base.haskey(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return haskey(gd, Tuple(key))
 end
 
+function Base.haskey(gd::GroupedDataFrame, key::AbstractDict{T, S}) where {T<:Union{Symbol, <:AbstractString}, S}
+    if length(key) != length(gd.cols) || any(Symbol(n) ∉ _names(gd) for n in keys(key))
```

Comment:
this test will not be needed if you move it inside `_dict_to_tuple`.

Reply:
Yes. all gone noe.